docs: Give more information about comparison report (#1871)
probabl-ai/skore: sphinx/user_guide/reporters.rst

Comparison report
-----------------

To compare the performance of different predictive models, `skore` provides the
:class:`ComparisonReport`.

:class:`ComparisonReport` takes a list (or a dictionary) of :class:`EstimatorReport` or
:class:`CrossValidationReport` instances. It then provides methods to compare the
performance of the different models.

In order for the comparison to make sense, the reports must all have the same test target.
However, they may have different training data or target; this might be the case when comparing a new model with the current production model, for example.
They may also have different testing data ($X_\text{test}$), which means the compared model pipelines do not necessarily need to be the same.

> This comparison of testing target is done thanks to hashing of the table. Therefore, if

Reviewer: ```suggestion
The comparison of testing targets is done by computing a hash of the arrays. Therefore, if
```

Author: I'm not sure about arrays / table, because it can be a multi-dim in case of multi-output. Even though by "excès de langage", indeed, we always talk about arrays.

same testing X, to remove the constraints of having consistent pipeline input.

> This comparison of testing target is done thanks to hashing of the table.

Reviewer: Do you mean that this is the method we'll use to check if the `y_test` are all the same? If so, this seems like an implementation detail; either remove it or add some detail on why that's important (e.g. I guess, that if the `y_test` are functionally equal, but have different data types, then the check will fail)

Author: You're right, we don't really need to know how the check is done actually.  
I don't know how hashing works exactly, "y_test are functionally equal, but have different data types, then the check will fail" --> was this just an example, or is it true?  For instance if I have one y with 1 and 0, and the other with 1.0 and 0.0 (on the same lines), does it output the same hash?

Reviewer: This is true, the two arrays in your example don't have the same hash

Reviewer: See https://github.com/probabl-ai/skore/blob/main/skore/src/skore/sklearn/_comparison/report.py#L174-L178
